File: Data Structures & Algorithms/majority-element-ii/submission-2.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int target=nums.length/3;
        Map<Integer, Integer> map = new HashMap<>();
        for(int num:nums){
            map.merge(num,1,Integer::sum);
        }
        List<Integer> ans=new ArrayList<>();
        for (Map.Entry<Integer,Integer> e: map.entrySet()){
            if (e.getValue()>target){
                ans.add(e.getKey());
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/majority-element/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        target=len(nums)//2
        curr=nums[0]
        cnt=0
        for num in nums:
            if num==curr:
                cnt+=1
            else:
                cnt-=1
            if cnt<=0:
                curr=num
                cnt=1
        return curr